### PR TITLE
First Draft

### PR DESCRIPTION
**Pull request by AnnJohn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,56 +3,145 @@
...
+  alert("Song Name: " + songFromAPIRequest.name);
+};
+
+$.getJSON(url, success_callback);
 ```
+We made a request to the API url and in our callback we created an alert dialog with the name of the song we got back. JQuery handled the request plus the parsing the result so all we need to do is to figure out what to do with the data.
 
-## Instructions
+## Changing Our Request With Parameters
+At some point, we will need to alter our request in order to change the data we receive. How we do this depends on the API we are calling but there is a good chance we will need to use url parameters. JQuery provides an easy way to do this by passing our parameters as a JavaScript object. Let's us the Spotify API to search for a song. We will specify 3 different criteria, `q` will be the search string, `type` will be the type of the thing we want to find and `limit` will be how many results we want.
```
Comment (victhevenot) [inline README.md]: why would you want to change the data we receive? what purpose does that serve?

File: README.md [hunk 2]
```diff
@@ -3,56 +3,145 @@
...
++ Make an API post request using AJAX
 + Parse JSON
 + Explain what JSON is
 
-
 ## Intro
 
-Now that we've made `get` requests to a website for text, let's do something more interesting with AJAX, let's use it to access an API.
-
+Now that we've made `get` requests to our server for text files, let's do something more interesting with AJAX by using it to access an API. Before we do that, let's talk about JSON.
```
Comment (victhevenot) [inline README.md]: Have intro talk more about the pain of dealing with data that isn't formatted nicely in JSON. want to prove the pain to students before we provide the easy-peasy solution that JSON provides. Something like it's hard to parse through a long string of data, etc.

File: README.md [hunk 3]
```diff
@@ -3,56 +3,145 @@
...
+  var tracks = searchResultsFromAPIRequest.tracks.items;
+  $.each(tracks, function(index, track) {
+    console.log("Song Found: " + track.name);
+  });
+};
+
+// No url parameters being passed since we already included them in the url
+$.getJSON(url, success_callback);
+```
+In the end, we get the same results but we have the option to choose our approach based on the situation. One final thing to consider is that our success callback won't actually be called if the JSON that gets return from the API is invalid. We will need to have a error callback to handle these types of problems.
```
Comment (victhevenot) [inline README.md:96]: can you name real world situations where you would want to manually create the URL versus pass in the url_parameters to the getJSON method?

File: README.md [hunk 4]
```diff
@@ -3,56 +3,145 @@
...
 
+## Instructions
```
Comment (victhevenot) [inline README.md:98]: Not a lot in this section for students to actually do themselves. It's definitely good to see how things work, but maybe leave some sections in each function that students need to fill in themselves.

File: README.md [hunk 5]
```diff
@@ -3,56 +3,145 @@
...
+        addHTML(response);
+      });
+    });
+  };
+  
+  $(document).ready(function(){
+    bindCreateButton();
+  });
+```
+Here we are sending markdown to the GitHub API to render into HTML. Once we get a response it adds the HTML to `<div id="search_results">`.
```
Comment (victhevenot) [inline README.md:149]: unclear what the form is supposed to do on the page. anything I entered  in the form didn't show up on the page because we hardcoded `#Hello World!`

also unclear how this ties in the github api. Visiting `https://api.github.com/markdown` in browser gives me a 404

File: README.md [hunk 6]
```diff
@@ -3,56 +3,145 @@
...
+  });
+};
+
+// No url parameters being passed since we already included them in the url
+$.getJSON(url, success_callback);
+```
+In the end, we get the same results but we have the option to choose our approach based on the situation. One final thing to consider is that our success callback won't actually be called if the JSON that gets return from the API is invalid. We will need to have a error callback to handle these types of problems.
 
+## Instructions
+You will need to visit https://github.com/settings/tokens and create a personal token to use before we begin. This token allows you to make requests for data related to your account. Once you have this token we can continue.
```
Comment (victhevenot) [inline README.md]: Not sure where to copy and paste my token in order to use the api. everything worked with just copying and pasting your code, so what is the point of the API token if my application isn't using it?

**Comment by victhevenot** [comment]
fixed some typos and left some comments!

File: README.md (hunk 4 above)
Comment (cbdileo) [inline README.md:98]: You're right but this was according to the outline provided. It said to create a code along where students copy and past code piece by piece and their are tests in the end to verify.

File: README.md (hunk 6 above)
Comment (cbdileo) [inline README.md]: You're right, I changed my example to not use the token to simplify but forgot to remove this.

File: README.md (hunk 5 above)
Comment (cbdileo) [inline README.md]: Typo thank you for pointing it out

Comment (cbdileo) [inline README.md:149]: I think this is a typo on my part

File: README.md (hunk 3 above)
Comment (cbdileo) [inline README.md:96]: The only reason would be that we aren't using jQuery but I figure this isn't a valid statement for this lesson. What do you think.

File: README.md (hunk 5 above)
Comment (cbdileo) [inline README.md:149]: As for your 404, that's because `https://api.github.com/markdown` only accepts POST requests

**Comment by AnnJohn** [comment]
assigned  back to @vicfriedman for final review

File: README.md (hunk 5 above)
Comment (victhevenot) [inline README.md:149]: awesome - makes total sense. maybe add in some expo about the API endpoint then, that you can't visit the route in the browser because it only accepts POSTs just so students don't have the same experience i did

File: README.md (hunk 4 above)
Comment (victhevenot) [inline README.md:98]: cool. leave it like that then. that's totally fine!
